src/MizarModels/include/MizarModels/FrameTrackListModel.h

```diff
@@ -0,0 +1,136 @@
+#include "MizarBase/Time.h"
+#include "MizarData/FrameTrack.h"
+#include "OrbitBase/Overloaded.h"
+#include "OrbitBase/Sort.h"
+
+Q_DECLARE_METATYPE(::orbit_mizar_data::FrameTrackId);
+
+namespace orbit_mizar_models {
+
+constexpr int kFrameTrackIdRole = Qt::UserRole + 1;
+
+// TODO(b/242839382) Add a dedicated unit test. Currently it is covered by the
+// SamplingWithFrameTrackInputWidgetTest.
+template <typename PairedData, typename FrameTrackStats>
+class FrameTrackListModelTmpl : public QAbstractListModel {
+  using FrameTrackId = ::orbit_mizar_data::FrameTrackId;
+  using FrameTrackInfo = ::orbit_mizar_data::FrameTrackInfo;
+  using RelativeTimeNs = ::orbit_mizar_base::RelativeTimeNs;
+  using TID = ::orbit_mizar_base::TID;
+  using PresentEvent = ::orbit_grpc_protos::PresentEvent;
+
+ public:
+  FrameTrackListModelTmpl(const PairedData* data, const absl::flat_hash_set<TID>* selected_tids,
+                          const RelativeTimeNs* start_timestamp, QObject* parent = nullptr)
+      : QAbstractListModel(parent),
+        data_(data),
+        selected_tids_(selected_tids),
+        start_timestamp_(start_timestamp),
+        id_to_displayed_name_(MakeDisplayedNames(data)) {}
+
+  [[nodiscard]] int rowCount(const QModelIndex& /*parent*/ = {}) const override {
+    return id_to_displayed_name_.size();
+  }
+
+  [[nodiscard]] QVariant data(const QModelIndex& index, int role = Qt::DisplayRole) const override {
+    if (index.model() != this) return {};
+    const auto& [id, name] = id_to_displayed_name_[index.row()];
+    switch (role) {
+      case Qt::DisplayRole:
+        return QString::fromStdString(name);
+      case Qt::ToolTipRole:
+        return MakeTooltip(id, name);
+      case kFrameTrackIdRole:
+        return QVariant::fromValue(id);
+      default:
+        return {};
+    }
+  }
+
+ private:
+  static std::vector<std::pair<FrameTrackId, std::string>> MakeDisplayedNames(
+      const PairedData* data) {
+    absl::flat_hash_map<FrameTrackId, FrameTrackInfo> id_to_infos = data->GetFrameTracks();
+    std::vector<std::pair<FrameTrackId, std::string>> id_to_displayed_name;
+    std::transform(std::begin(id_to_infos), std::end(id_to_infos),
+                   std::back_inserter(id_to_displayed_name), [](const auto& id_to_info) {
+                     const auto [id, info] = id_to_info;
+                     const std::string displayed_name = MakeDisplayedName(info);
+
+                     return std::make_pair(id, displayed_name);
+                   });
+
+    orbit_base::sort(std::begin(id_to_displayed_name), std::end(id_to_displayed_name),
+                     &std::pair<FrameTrackId, std::string>::second);
+    return id_to_displayed_name;
+  }
+
+  [[nodiscard]] static std::string MakeFrameTrackString(
+      const orbit_client_data::ScopeInfo& scope_info) {
+    const std::string_view type_string =
+        scope_info.GetType() == orbit_client_data::ScopeType::kDynamicallyInstrumentedFunction
+            ? "  D"
```

Comment:
Nit: maybe define some constants for this, which will also benefit the unit tests

Reply:
But we would have to expose the constants and write some more code in the test glueing together these prefixes and the names (for now I've just got this hard-coded in the test and IMO it's fine).